lib/layout/flow.dart

```diff
@@ -105,24 +106,34 @@ class FlowState extends WidgetState<Flow> with TemplatedWidgetState {
       children.addAll(templatedChildren!);
     }
 
-    Widget rtn = Wrap(
-      direction: widget._controller.direction == Axis.vertical.name ? Axis.vertical : Axis.horizontal,
-      spacing: widget.controller.gap?.toDouble() ?? 0,
-      runSpacing: widget._controller.lineGap?.toDouble() ?? 0,
-      alignment: widget._controller.mainAxis ?? WrapAlignment.start,
-      crossAxisAlignment: widget._controller.crossAxis ?? WrapCrossAlignment.start,
-      children: children,
+    Widget rtn = BoxWrapper(
+      boxController: widget._controller,
+      ignoresMargin: true,
+      widget: Wrap(
+            direction: widget._controller.direction == Axis.vertical.name ? Axis.vertical : Axis.horizontal,
+            spacing: widget.controller.gap?.toDouble() ?? 0,
+            runSpacing: widget._controller.lineGap?.toDouble() ?? 0,
+            alignment: widget._controller.mainAxis ?? WrapAlignment.start,
+            crossAxisAlignment: widget._controller.crossAxis ?? WrapCrossAlignment.start,
+            children: children,
+          ),
     );
 
-    if (widget._controller.maxWidth != null || widget._controller.maxHeight != null) {
+    if (widget._controller.maxWidth != null || widget._controller.maxHeight != null || widget._controller.margin!=null) {
```

Comment:
Don't wrap the widget inside a ConstrainedBox just because you have a margin. That's another widget you have to render. Do that before you get to this block.